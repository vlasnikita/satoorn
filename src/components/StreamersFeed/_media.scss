@media only screen and (max-width: 800px) {
    .Streamer{
        width: 380px;
        padding: 10px 20px;
        height: 70px;
        min-height: 70px;
    }
}

@media only screen and (max-width: 600px) {
    .Streamer{
        width: 270px;
        padding: 6px 12px;

        .UserInfo{
            padding-left: 0;
            width: 100%;
        }

        .Button_wide{
            width: 90px;
        }
    }
}