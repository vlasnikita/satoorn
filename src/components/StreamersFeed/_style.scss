.StreamersFeed{
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    overflow: scroll;
    width: 100%;
    height: calc(100% - 50px);
    padding-top: $headerHeight;
}

.Streamer{
    text-decoration: none;
    width: 440px;
    border-radius: 5px;
    position: relative;
    background: #fff;
    padding: 15px 30px;
    margin: 15px 0;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    box-shadow: 0 0 10px 1px rgba(17, 17, 17, .125);
    cursor: pointer;
    overflow: hidden;
    height: 80px;
    min-height: 80px;

    &__avatar {
        transition: .25s ease-out;
        border-radius: 50%;
        width: 80px;
        height: 80px;
        margin-right: 20px;
        border: 3px solid #fff;
        box-shadow: 0 0 15px 2px rgba(0, 0, 0, 0.1);
    }

}

.StreamersFeed::-webkit-scrollbar-track {
display: none;
}

.StreamersFeed::-webkit-scrollbar {
display: none;
}

.StreamersFeed::-webkit-scrollbar-thumb {
display: none;
}


//$loadingBg: #f8f8f8;
.Streamer_loading {
opacity: .75;
cursor: auto;

    &:hover {
        &:after {
            display: none;
        }
    }

    .Streamer__avatar {
        box-shadow: none;
    }


    .Streamer__button{
        box-shadow: none;
    }
}

@import "/media";