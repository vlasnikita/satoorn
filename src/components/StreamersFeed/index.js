import React, {Component, Fragment} from 'react'
import {connect} from 'react-redux'
import {Link} from "react-router-dom"
import {getMoreStreamers, getStreamers} from "Actions/streamers"
import finishIcon from 'Static/finish.png'
import {logUserAction} from 'Utils/metrics';
import UserInfo from "Components/UserInfo/";

class StreamersFeed extends Component {

    state = {
        isFirstLoading: true
    };

    componentDidMount() {
        document.title = 'Стримеры'

        if (this.props.searchTerm.length > 1) this.props.handleGetStreamers(this.props.searchTerm);
        else this.props.handleGetStreamers();

        const div = document.querySelector('.StreamersFeed');
        if(!!div){
            // подгрузка данных при доскролливании до низа страницы
            div.addEventListener('scroll', this.handleScroll)
        }
    }

    componentWillUpdate() {
        if (this.state.isFirstLoading && this.props.isStreamersLoaded) this.setState({isFirstLoading: false})
    }

    componentDidUpdate() {
        if (this.props.isFiltersUpdated) {
            if (this.props.searchTerm.length > 1) this.props.handleGetStreamers(this.props.searchTerm);
            else this.props.handleGetStreamers()
        }
    }

    handleScroll = e => {
        const div = e.target
        if (!this.props.isStreamersNoMore && !this.props.isStreamersLoading && div.scrollTop + div.clientHeight > div.scrollHeight - 200) {
            if (this.props.searchTerm.length > 1) this.props.handleGetMoreStreamers(this.props.streamers.length + 1, this.props.searchTerm);
            else this.props.handleGetMoreStreamers(this.props.streamers.length + 1, this.props.searchTerm)
        }
    }

    getFirstLoadingScreen = () => (
        <div className="StreamersFeed">
            {'12345678'.split('').map(i => (
                <div className="Streamer Streamer_loading" key={i}>
                    <span className="Streamer__avatar loader"/>
                    <div className="Streamer__info">
                        <div className="Streamer__header loader">
                            <span className="Streamer__name"/>
                        </div>
                        <div className="Streamer__crumb loader"/>
                    </div>
                    <span className="Streamer__button loader"/>
                </div>
            ))}
        </div>
    );

    getLoadingFooter = () => {
        if (this.state.isFirstLoading) return null;
        else if (this.props.isStreamersNoMore) return (
            <Fragment>
                <span className="loader__finishBefore"/>
                <div className="loader__finish">
                    <span className="loader__finishCircle"/>
                    <img src={finishIcon} alt={"loading..."}/>
                </div>
            </Fragment>
        );
        else if (this.props.isStreamersLoading) return (
            <div className="loader__pending">
                <span className="loader__pendingDot"/>
                <div className="loader__pendingDots"><span/><span/><span/></div>
            </div>
        );
        else return null
    };

    render() {
        if (this.state.isFirstLoading && this.props.isStreamersLoading) return this.getFirstLoadingScreen();
        return (
            <div className="StreamersFeed">
                {Array.isArray(this.props.streamers) && this.props.streamers.map((streamer, i) => (
                    <Link
                        to={`/${streamer.login}`}
                        className="Streamer" key={i}
                    >
                        <UserInfo feed streamer={streamer}/>
                        <Link
                            id="Streamer__button"
                            onClick={() => this.logEventResolvingCurrentProfile()}
                            className="Button Button_wide"
                            to={`/${streamer.login}/`}
                        >Перейти в профиль
                        </Link>
                    </Link>
                ))}
                {this.getLoadingFooter()}
            </div>
        );
    }

    logEventResolvingCurrentProfile() {
        const userId = !!this.props.profile && !!this.props.profile.id ? this.props.profile.id : null;
      logUserAction('STREAMERS_FEED', 'CREATE_CHALLENGE', userId)
    }
}

const mapStateToProps = state => ({
    profile: state.profile.profile,
    streamers: state.streamers.streamers,
    isStreamersLoading: state.streamers.loading,
    isStreamersLoaded: state.streamers.loaded,
    isStreamersNoMore: state.streamers.noMore,
    searchTerm: state.streamers.searchTerm,
    isFiltersUpdated: state.streamers.filtersUpdated,
});

const mapDispatchToProps = dispatch => {
    return {
        handleGetStreamers: searchTerm => dispatch(getStreamers(searchTerm)),
        handleGetMoreStreamers: (from, searchTerm) => dispatch(getMoreStreamers(from, searchTerm))
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(StreamersFeed);
