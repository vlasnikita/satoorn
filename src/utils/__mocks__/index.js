const utils = jest.requireActual('Utils/index')

export const getOperationHash = jest.fn().mockImplementation(str => 'MOCKED_HASH')

export const getDonatersEmoji = jest.fn().mockImplementation(() => utils.getDonatersEmoji())

export const linkifyPlainText = jest.fn().mockImplementation((text) => utils.linkifyPlainText(text))

export const openPaymentWidget =  jest.fn().mockImplementation((amount, transactionId, desc, isDonation, challengeId, challengeBody) =>{})