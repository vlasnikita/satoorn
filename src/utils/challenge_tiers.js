import venzelGreenIcon from "Static/challenge/venzel-green.svg";
import venzelBlueIcon from "Static/challenge/venzel-blue.svg";
import venzelPurpleIcon from "Static/challenge/venzel-purple.svg";
import venzelYellowIcon from "Static/challenge/venzel-yellow.svg";
import React from "react";

const TIER_LIMITS_RUR = {
    t5: 0,
    t4: 150,
    t3: 300,
    t2: 700,
    t1: 1000
}

export const getTierNumber = donationAmountString => {
    const donationAmount = +donationAmountString

    if(donationAmount < TIER_LIMITS_RUR.t4) return 5
    else if(donationAmount < TIER_LIMITS_RUR.t3) return 4
    else if(donationAmount < TIER_LIMITS_RUR.t2) return 3
    else if(donationAmount < TIER_LIMITS_RUR.t1) return 2
    else return 1
}

export const getVenzel = donationAmountString => {
    const donationAmount = +donationAmountString
    const tier = getTierNumber(donationAmount);

    if (tier === 4) return <img className='Challenge__venzel' src={venzelGreenIcon} alt=""/>;
    else if (tier === 3) return <img className='Challenge__venzel' src={venzelBlueIcon} alt=""/>;
    else if (tier === 2) return <img className='Challenge__venzel' src={venzelPurpleIcon} alt=""/>;
    else if (tier === 1) return <img className='Challenge__venzel' src={venzelYellowIcon} alt=""/>
};