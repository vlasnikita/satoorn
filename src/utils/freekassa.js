import md5 from 'md5'

const merchant_id = 17257
const secret = 'lqEKapouwJ4MUBH'

export function generateFKPaymentLink (sum, challenge_id){
    const url_base = 'https://pay.freekassa.ru/'
    const currency = 'RUB'
    const md5_input = `${merchant_id}:${sum}:${secret}:${currency}:${challenge_id}`
    const signature = md5(md5_input)
    
    const url = `${url_base}?m=${merchant_id}&oa=${sum}&currency=${currency}&o=${challenge_id}&s=${signature}`
    return url
}