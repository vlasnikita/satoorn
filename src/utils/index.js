import {UnitPay} from 'Static/js/unitpay'
import {YooMoneyCheckout} from 'Static/js/yoomoney'
import sha256 from 'crypto-js/sha256'
import QRCode  from 'qrcode';

import store from 'Store/index';
import { SECRET_KEY, PUBLIC_KEY, HOST } from 'Constants/payment_module'
import { HOSTING_ADDRESS } from 'Constants/routes'
import ALERT_TYPE from 'Constants/alert_type'

import { setActiveTransactionSuccess, setActiveTransactionFail } from 'Actions/transactions'
import { setRedirectToChallenge } from 'Actions/challenges'
import { updateChallenge } from 'Actions/challenges'
import { showAlert } from 'Actions/alert'
import { getChallengeById } from 'Actions/challenges';
import log from './logging';

export const validateEmail = email => {
    const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return re.test(String(email).toLowerCase());
}

export const validateUrl = url => {
    const re = /(((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:)*@)?(((\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5]))|((([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.)+(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.?)(:\d*)?)(\/((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)+(\/(([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)*)*)?)?(\?((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)|[\uE000-\uF8FF]|\/|\?)*)?(\#((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)|\/|\?)*)?$/
    return re.test(String(url).toLowerCase());
}

export const concatenateUrlParams = paramsMap => {
    let urlParamsString = ''
    const isObject = paramsMap === Object(paramsMap) && Object.prototype.toString.call(paramsMap) !== '[object Array]'
    if(!paramsMap || !isObject) return urlParamsString

    const keys = Object.keys(paramsMap)
    if(keys.length === 0) return urlParamsString
    else {
        Object.keys(paramsMap).forEach(paramKey => {
            if(!!paramsMap[paramKey]) {
                if (urlParamsString.length === 0) urlParamsString = `?${paramKey}=${paramsMap[paramKey]}`
                else urlParamsString = `${urlParamsString}&${paramKey}=${paramsMap[paramKey]}`
            }
        })
        return urlParamsString
    }
}

export const getFormSignature = (amount, transaction, desc) => {
    const hashStr = `${transaction}{up}${desc}{up}${amount}{up}${SECRET_KEY}`
    return sha256(hashStr)
}

export const openInNewTab = url => {
    var win = window.open(url, '_blank');
    win.focus();
}

export const openPaymentWidget = (amount, transaction, desc, isDonation, challengeId, challengeBody, streamerLogin) => {
    const signature = getFormSignature(amount, transaction, desc)

    // store.dispatch(getChallengeById(challengeId))

    const checkout = new window.YooMoneyCheckoutWidget({
        confirmation_token: 'confirmation-token', //TODO set proper token
        return_url: 'https://satoorn.ru/', //TODO set redirect 200 url
        customization: {
            //TODO customize view here
            modal: true
        },
        error_callback: function(error) {
            console.log(error)
        }
    });
  
    checkout.on('modal_close', () => {
        console.log('Код, который нужно выполнить после закрытия всплывающего окна.')
    });

    checkout.on('success', () => {
        log.debug('Успешный платеж', params);

        if(isDonation) {
            store.dispatch(updateChallenge(challengeId, {...challengeBody, isJustUpdated: true}))
        }
        else {
            store.dispatch(setActiveTransactionSuccess())
            store.dispatch(setRedirectToChallenge(challengeId, streamerLogin))
        }
        store.dispatch(showAlert(ALERT_TYPE.OK))

        checkout.destroy();
    });

    checkout.on('fail', () => {
        log.debug(message, params);

        store.dispatch(setActiveTransactionFail())
        store.dispatch(showAlert(ALERT_TYPE.PS_ERROR))

        checkout.destroy();
    });

    //Отображение платежной формы в контейнере
    checkout.render('payment-form');


    /*
    ----- OBSOLETE UNITPAY, EXAMPLE PURPOSES -----
    */
    // const payment = new UnitPay();
    // payment.createWidget({
    //     sum: +amount,
    //     desc: desc.toString(),
    //     account: transaction.toString(),
    //     domainName: "unitpay.money",
    //     signature: signature.toString(),
    //     publicKey: PUBLIC_KEY,
    //     locale: 'ru'
    // });

    // payment.success(function (params) {
    //     log.debug('Успешный платеж', params);

    //     if(isDonation) {
    //         store.dispatch(updateChallenge(challengeId, {...challengeBody, isJustUpdated: true}))
    //     }
    //     else {
    //         store.dispatch(setActiveTransactionSuccess())
    //         store.dispatch(setRedirectToChallenge(challengeId, streamerLogin))
    //     }
    //     store.dispatch(showAlert(ALERT_TYPE.OK))
    // });
    // payment.error(function (message, params) {
    //     log.debug(message, params);

    //     store.dispatch(setActiveTransactionFail())
    //     store.dispatch(showAlert(ALERT_TYPE.PS_ERROR))
    // });
    // return false;
}

export const getDonatersEmoji = () => {
    if(process.env.NODE_ENV === 'test') return ''
    const emojis = ['😱', '😵', '🤩', '🤗','🤔', '😮', '😌', '🙃']
    return emojis[ Math.floor(Math.random() * emojis.length) ]
}

export const getAlertSuccessEmoji = () => {
    if(process.env.NODE_ENV === 'test') return ''
    const emojis = [
        '😊','😋','😉','️😏','😌','🤤', '😸','️🙌',
        '👍','👊','✊','✌', '️🤘','🤟','👌','🤙',
        '🐒','🙈', '🌝','🎉','🎊','🔥','✨']

    return emojis[ Math.floor(Math.random() * emojis.length) ]

}

export const getOperationHash = str => {
    return sha256(str).toString()
};

export const linkifyPlainText = text => text.replace(/((http:|https:)[^\s]+[\w])/g, '<a href="$1" target="_blank">$1</a>')

export const checkIfNoOuterUIPage = () => {
    return (
        window.location.pathname === '/policy' ||
        window.location.pathname === '/agreement'
    )
}

export function onlyUnique(value, index, self) {
    return self.indexOf(value) === index;
}

export const getItemByKey = (array, key) => array.reduce((acc, item) => {
    if(!acc) return item.key === key ? item : null
    else return acc
}, null)

export const generateQRCode = async text => {
    try {
      return await QRCode.toDataURL(text)
    } catch (err) {
      console.error(err)
    }
}