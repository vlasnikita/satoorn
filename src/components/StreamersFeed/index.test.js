import {configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import 'jest-specific-snapshot';
import React from 'react';
import {render, unmountComponentAtNode} from "react-dom";
import {act} from "react-dom/test-utils";
import {Provider} from "react-redux";
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureMockStore from "redux-mock-store";
import {defaultTestInitialState} from '../../test/utils/commonTestUtils';
import StreamersFeed from './index';

jest.mock('Utils/metrics');

configure({adapter: new Adapter()});
const streamers = require('Mocks/streamers');
const mockStore = configureMockStore();
const streamer = streamers[0];

it('renders correctly', () => {
    const store = mockStore(defaultTestInitialState);
    const tree = renderer.create(
        <BrowserRouter>
            <Provider store={store}>
                <StreamersFeed
                />
            </Provider>
        </BrowserRouter>
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('renders correctly not empty streamers feed limited by scroll position', () => {
    let parent = prepareParent();

    const store = mockStore(defaultTestInitialState);
    act(() => {
        render(
            <BrowserRouter>
                <Provider store={store}>
                    <StreamersFeed/>
                </Provider>
            </BrowserRouter>,
            parent);
    });

    expect(parent.querySelectorAll('.Streamer').length).toEqual(defaultTestInitialState.streamers.streamers.length);

    cleanupParent(parent);
});


function clickOnGoToProfileButton(feed) {
    feed.findWhere(node => node.type() === 'a'
        &&
        node.hasClass('Button')
    ).simulate('click');
}

function cleanupParent(parent) {
    unmountComponentAtNode(parent);
    parent.remove();
}

function prepareParent() {
    let parent = document.createElement("div");
    document.body.appendChild(parent);
    return parent;
}

