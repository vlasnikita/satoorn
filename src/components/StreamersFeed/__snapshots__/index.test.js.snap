// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="StreamersFeed"
>
  <a
    className="Streamer"
    href="/user_1"
    onClick={[Function]}
  >
    <div
      className="UserInfo"
    >
      <img
        className="UserInfo__avatarPipe"
        src="user-create-pipe.svg"
      />
      <img
        alt="N/A"
        className="UserInfo__avatar"
        src="https://images-cdn.9gag.com/photo/anbZVN0_700b.jpg"
      />
      <div
        className="UserInfo__rightColumn"
      >
        <h4
          className="UserInfo__login"
        >
          user_1
        </h4>
        <div
          className="UserInfo__details"
        >
          <div
            className="UserInfo__crumb"
          >
            <p
              className="UserInfo__crumb__text"
            >
              12 челленджей
            </p>
          </div>
        </div>
      </div>
    </div>
    <a
      className="Button Button_wide"
      href="/user_1/"
      id="Streamer__button"
      onClick={[Function]}
    >
      Перейти в профиль
    </a>
  </a>
  <a
    className="Streamer"
    href="/user_2"
    onClick={[Function]}
  >
    <div
      className="UserInfo"
    >
      <img
        className="UserInfo__avatarPipe"
        src="user-create-pipe.svg"
      />
      <img
        alt="N/A"
        className="UserInfo__avatar"
        src="https://images-cdn.9gag.com/photo/anbZVN0_700b.jpg"
      />
      <div
        className="UserInfo__rightColumn"
      >
        <h4
          className="UserInfo__login"
        >
          user_2
        </h4>
        <div
          className="UserInfo__details"
        >
          <div
            className="UserInfo__crumb"
          >
            <p
              className="UserInfo__crumb__text"
            >
              69 челленджей
            </p>
          </div>
        </div>
      </div>
    </div>
    <a
      className="Button Button_wide"
      href="/user_2/"
      id="Streamer__button"
      onClick={[Function]}
    >
      Перейти в профиль
    </a>
  </a>
  <a
    className="Streamer"
    href="/user_4"
    onClick={[Function]}
  >
    <div
      className="UserInfo"
    >
      <img
        className="UserInfo__avatarPipe"
        src="user-create-pipe.svg"
      />
      <img
        alt="N/A"
        className="UserInfo__avatar"
        src="https://images-cdn.9gag.com/photo/anbZVN0_700b.jpg"
      />
      <div
        className="UserInfo__rightColumn"
      >
        <h4
          className="UserInfo__login"
        >
          user_4
        </h4>
        <div
          className="UserInfo__details"
        >
          <div
            className="UserInfo__crumb"
          >
            <p
              className="UserInfo__crumb__text"
            >
              69 челленджей
            </p>
          </div>
        </div>
      </div>
    </div>
    <a
      className="Button Button_wide"
      href="/user_4/"
      id="Streamer__button"
      onClick={[Function]}
    >
      Перейти в профиль
    </a>
  </a>
  <a
    className="Streamer"
    href="/user_5"
    onClick={[Function]}
  >
    <div
      className="UserInfo"
    >
      <img
        className="UserInfo__avatarPipe"
        src="user-create-pipe.svg"
      />
      <img
        alt="N/A"
        className="UserInfo__avatar"
        src="https://images-cdn.9gag.com/photo/anbZVN0_700b.jpg"
      />
      <div
        className="UserInfo__rightColumn"
      >
        <h4
          className="UserInfo__login"
        >
          user_5
        </h4>
        <div
          className="UserInfo__details"
        >
          <div
            className="UserInfo__crumb"
          >
            <p
              className="UserInfo__crumb__text"
            >
              69 челленджей
            </p>
          </div>
        </div>
      </div>
    </div>
    <a
      className="Button Button_wide"
      href="/user_5/"
      id="Streamer__button"
      onClick={[Function]}
    >
      Перейти в профиль
    </a>
  </a>
  <a
    className="Streamer"
    href="/user_7"
    onClick={[Function]}
  >
    <div
      className="UserInfo"
    >
      <img
        className="UserInfo__avatarPipe"
        src="user-create-pipe.svg"
      />
      <img
        alt="N/A"
        className="UserInfo__avatar"
        src="https://images-cdn.9gag.com/photo/anbZVN0_700b.jpg"
      />
      <div
        className="UserInfo__rightColumn"
      >
        <h4
          className="UserInfo__login"
        >
          user_7
        </h4>
        <div
          className="UserInfo__details"
        >
          <div
            className="UserInfo__crumb"
          >
            <p
              className="UserInfo__crumb__text"
            >
              69 челленджей
            </p>
          </div>
        </div>
      </div>
    </div>
    <a
      className="Button Button_wide"
      href="/user_7/"
      id="Streamer__button"
      onClick={[Function]}
    >
      Перейти в профиль
    </a>
  </a>
  <a
    className="Streamer"
    href="/user_800"
    onClick={[Function]}
  >
    <div
      className="UserInfo"
    >
      <img
        className="UserInfo__avatarPipe"
        src="user-create-pipe.svg"
      />
      <img
        alt="N/A"
        className="UserInfo__avatar"
        src="https://images-cdn.9gag.com/photo/anbZVN0_700b.jpg"
      />
      <div
        className="UserInfo__rightColumn"
      >
        <h4
          className="UserInfo__login"
        >
          user_800
        </h4>
        <div
          className="UserInfo__details"
        >
          <div
            className="UserInfo__crumb"
          >
            <p
              className="UserInfo__crumb__text"
            >
              69 челленджей
            </p>
          </div>
        </div>
      </div>
    </div>
    <a
      className="Button Button_wide"
      href="/user_800/"
      id="Streamer__button"
      onClick={[Function]}
    >
      Перейти в профиль
    </a>
  </a>
  <a
    className="Streamer"
    href="/user_802"
    onClick={[Function]}
  >
    <div
      className="UserInfo"
    >
      <img
        className="UserInfo__avatarPipe"
        src="user-create-pipe.svg"
      />
      <img
        alt="N/A"
        className="UserInfo__avatar"
        src="https://images-cdn.9gag.com/photo/anbZVN0_700b.jpg"
      />
      <div
        className="UserInfo__rightColumn"
      >
        <h4
          className="UserInfo__login"
        >
          user_802
        </h4>
        <div
          className="UserInfo__details"
        >
          <div
            className="UserInfo__crumb"
          >
            <p
              className="UserInfo__crumb__text"
            >
              69 челленджей
            </p>
          </div>
        </div>
      </div>
    </div>
    <a
      className="Button Button_wide"
      href="/user_802/"
      id="Streamer__button"
      onClick={[Function]}
    >
      Перейти в профиль
    </a>
  </a>
  <a
    className="Streamer"
    href="/user_803"
    onClick={[Function]}
  >
    <div
      className="UserInfo"
    >
      <img
        className="UserInfo__avatarPipe"
        src="user-create-pipe.svg"
      />
      <img
        alt="N/A"
        className="UserInfo__avatar"
        src="https://images-cdn.9gag.com/photo/anbZVN0_700b.jpg"
      />
      <div
        className="UserInfo__rightColumn"
      >
        <h4
          className="UserInfo__login"
        >
          user_803
        </h4>
        <div
          className="UserInfo__details"
        >
          <div
            className="UserInfo__crumb"
          >
            <p
              className="UserInfo__crumb__text"
            >
              69 челленджей
            </p>
          </div>
        </div>
      </div>
    </div>
    <a
      className="Button Button_wide"
      href="/user_803/"
      id="Streamer__button"
      onClick={[Function]}
    >
      Перейти в профиль
    </a>
  </a>
  <a
    className="Streamer"
    href="/user_808"
    onClick={[Function]}
  >
    <div
      className="UserInfo"
    >
      <img
        className="UserInfo__avatarPipe"
        src="user-create-pipe.svg"
      />
      <img
        alt="N/A"
        className="UserInfo__avatar"
        src="https://images-cdn.9gag.com/photo/anbZVN0_700b.jpg"
      />
      <div
        className="UserInfo__rightColumn"
      >
        <h4
          className="UserInfo__login"
        >
          user_808
        </h4>
        <div
          className="UserInfo__details"
        >
          <div
            className="UserInfo__crumb"
          >
            <p
              className="UserInfo__crumb__text"
            >
              69 челленджей
            </p>
          </div>
        </div>
      </div>
    </div>
    <a
      className="Button Button_wide"
      href="/user_808/"
      id="Streamer__button"
      onClick={[Function]}
    >
      Перейти в профиль
    </a>
  </a>
  <a
    className="Streamer"
    href="/user_1000"
    onClick={[Function]}
  >
    <div
      className="UserInfo"
    >
      <img
        className="UserInfo__avatarPipe"
        src="user-create-pipe.svg"
      />
      <img
        alt="N/A"
        className="UserInfo__avatar"
        src="https://images-cdn.9gag.com/photo/anbZVN0_700b.jpg"
      />
      <div
        className="UserInfo__rightColumn"
      >
        <h4
          className="UserInfo__login"
        >
          user_1000
        </h4>
        <div
          className="UserInfo__details"
        >
          <div
            className="UserInfo__crumb"
          >
            <p
              className="UserInfo__crumb__text"
            >
              69 челленджей
            </p>
          </div>
        </div>
      </div>
    </div>
    <a
      className="Button Button_wide"
      href="/user_1000/"
      id="Streamer__button"
      onClick={[Function]}
    >
      Перейти в профиль
    </a>
  </a>
  <a
    className="Streamer"
    href="/user_810"
    onClick={[Function]}
  >
    <div
      className="UserInfo"
    >
      <img
        className="UserInfo__avatarPipe"
        src="user-create-pipe.svg"
      />
      <img
        alt="N/A"
        className="UserInfo__avatar"
        src="https://images-cdn.9gag.com/photo/anbZVN0_700b.jpg"
      />
      <div
        className="UserInfo__rightColumn"
      >
        <h4
          className="UserInfo__login"
        >
          user_810
        </h4>
        <div
          className="UserInfo__details"
        >
          <div
            className="UserInfo__crumb"
          >
            <p
              className="UserInfo__crumb__text"
            >
              69 челленджей
            </p>
          </div>
        </div>
      </div>
    </div>
    <a
      className="Button Button_wide"
      href="/user_810/"
      id="Streamer__button"
      onClick={[Function]}
    >
      Перейти в профиль
    </a>
  </a>
</div>
`;
